src/components/inputs/time-input/time-input.js

```diff
@@ -55,14 +56,17 @@ const TimeInput = props => {
 
   // so that it doesn't run on initial mount
   const mounted = React.useRef();
-  React.useEffect(() => {
-    if (!mounted.current) {
-      mounted.current = true;
-    } else {
-      emitChange(TimeInput.toLocaleTime(value, intl.locale));
-    }
-  }, [intl.locale, emitChange, value]);
-
+  React.useEffect(
+    () => {
+      if (!mounted.current) {
+        mounted.current = true;
+      } else {
+        emitChange(TimeInput.toLocaleTime(value, intl.locale));
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
What's happening here?

Reply:
You're right, this isn't good. I was trying to call emitChange when the locale changed.

I changed the approach in the last commit to this

```js

  const prevLocale = usePrevious(intl.locale);

  // if locale has changed
  if (typeof prevLocale !== 'undefined' && prevLocale !== intl.locale) {
    emitChange(TimeInput.toLocaleTime(value, intl.locale));
  }

```